server/planning/search/planning_search.py

```diff
@@ -70,35 +71,26 @@ def get(self, req, lookup):
 
         if on_fetched_resource:
             for resource in types:
-                response = {app.config['ITEMS']: [doc for doc in docs if doc['_type'] == resource]}
+                response = {
+                    app.config['ITEMS']: [
+                        doc
+                        for doc in docs
+                        if doc['_type'] == resource
+                    ]
+                }
                 getattr(app, 'on_fetched_resource')(resource, response)
                 getattr(app, 'on_fetched_resource_%s' % resource)(response)
 
         return docs
 
-    def _get_resource_schema(self, resource):
-        datasource = self.elastic.get_datasource(resource)
-        schema = {}
-        schema.update(superdesk.config.DOMAIN[datasource[0]].get('schema', {}))
-        schema.update(superdesk.config.DOMAIN[resource].get('schema', {}))
-        return schema
-
-    def _parse_hits(self, hits):
-        schemas = {
-            'planning': self._get_resource_schema('planning'),
-            'events': self._get_resource_schema('events')
-        }
-
-        docs = []
-        for hit in hits.get('hits', {}).get('hits', []):
-            item_type = hit.get('_type')
-            schema = schemas.get(item_type)
-            dates = get_dates(schema)
-            doc = format_doc(hit, schema, dates)
-            build_custom_hateoas({'self': {'title': doc['_type'], 'href': '/{}/{{_id}}'.format(doc['_type'])}}, doc)
-            docs.append(doc)
-
-        return ElasticCursor(hits, docs)
+    def _format_nested_dates(self, doc):
+        if doc.get('_type') == 'events' and doc.get('dates'):
+            if doc['dates'].get('start'):
+                doc['dates']['start'] = parse_date(doc['dates']['start'])
+            if doc['dates']['end']:
```

Comment:
I think It would be good if you use `doc['dates'].get('end')` instead of `doc['dates']['end']`